koku/api/provider/serializers.py

```diff
@@ -318,6 +322,8 @@ def create(self, validated_data):
     def update(self, instance, validated_data):
         """Update a Provider instance from validated data."""
         provider_type = validated_data['type']
+        provider_type = Provider.PROVIDER_CASE_MAPPING.get(provider_type, provider_type)
```

Comment:
Should this be?
`provider_type = Provider.PROVIDER_CASE_MAPPING.get(provider_type.lower())`

and then throw a validation error if provider type is None?

Reply:
I did remove the default of provider_type, but I didn't introduce a new validation error - because I found that it was already being caught. For example, if I create a provider with the type being `notaprovider` then I get back this response from postman which seems reasonable to me: 
```
{
    "errors": [
        {
            "detail": "notaprovider is not a valid source type.",
            "source": "type",
            "status": 400
        }
    ]
}
```